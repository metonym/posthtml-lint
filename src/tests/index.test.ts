import posthtml from 'posthtml';
import { lint } from '../';
import { testCases } from './__fixtures__/testCases';

describe('posthtmlLint', () => {
  testCases.forEach(testCase => {
    test(`'${testCase.name}' matches the snapshot`, () => {
      posthtml()
        .use(lint())
        .process(testCase.input)
        .then((result: { html: string }) => {
          expect(result.html).toMatchSnapshot();
        });
    });
  });
});
