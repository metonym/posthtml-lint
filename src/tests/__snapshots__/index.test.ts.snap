// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`posthtmlLint 'No Duplicate Ids' matches the snapshot 1`] = `
"
      <div id=\\"a\\"></div>
      <div id=\\"a\\"></div>
    "
`;

exports[`posthtmlLint 'No Duplicate Tags' matches the snapshot 1`] = `
"
      <body></body>
      <body></body>
    "
`;

exports[`posthtmlLint 'No Empty Tags' matches the snapshot 1`] = `
"
      <a href=\\"\\"></a>
      <img>
      <div><span></span></div>
    "
`;

exports[`posthtmlLint 'No Missing Attributes' matches the snapshot 1`] = `
"
      <a></a>
      <img>
      <input>
    "
`;
